paasta_tools/cli/cmds/status.py

```diff
@@ -1236,13 +1256,167 @@ def get_version_table_entry(
     humanized_start_time = humanize.naturaltime(start_datetime)
     entry = [f"{version_name} - Started {start_datetime} ({humanized_start_time})"]
     replica_states = get_replica_states(replicaset.pods)
-    entry.append(f"  Replica States: {replica_states}")
+    replica_states = sorted(replica_states, key=lambda s: s[1].create_timestamp)
+
+    if replica_states:
+        # If no replica_states, there were no pods found
+        replica_state_list = []
+        for state in ReplicaState:
+            this_state_replicas = [rs for rs in replica_states if rs[0] == state]
+            if this_state_replicas:
+                replica_state_list.append(
+                    state.color(f"{len(this_state_replicas)} {state.value}")
+                )
+
+        entry.append(f"  Replica States: {' / '.join(replica_state_list)}")
+        if not verbose:
+            unhealthy_replicas = [rs for rs in replica_states if rs[0].is_unhealthy()]
+            if unhealthy_replicas:
+                entry.append(f"    Unhealthy Replicas:")
+                replica_table = create_replica_table(unhealthy_replicas)
+                for line in replica_table:
+                    entry.append(f"      {line}")
+        else:
+            replica_table = create_replica_table(replica_states)
+            for line in replica_table:
+                entry.append(f"    {line}")
     return entry
 
 
-# TODO(PAASTA-17287): Implement replica states
-def get_replica_states(pods: List[KubernetesPodV2]) -> str:
-    return PaastaColors.green(f"{len(pods)} Running")
+class ReplicaState(Enum):
+    RUNNING = "Healthy"
+    UNSCHEDULED = "Unscheduled"
+    STARTING = "Starting"
+    WARMING_UP = "Warming Up"
+    TERMINATING = "Terminating"
+    HEALTHCHECK_FAILING = "Healthcheck Failing"
+    UNREACHABLE = "Unreachable"
+    UNHEALTHY = "Unhealthy"
+    UNKNOWN = "Unknown"
+
+    def is_unhealthy(self):
+        return self in ReplicaState.unhealthy_states()
+
+    @property
+    def color(self):
+        if self.is_unhealthy():
+            return PaastaColors.red
+        elif self.name == "RUNNING":
+            return PaastaColors.green
+        return PaastaColors.yellow
+
+    @classmethod
+    def unhealthy_states(cls):
+        return [cls.UNHEALTHY, cls.UNREACHABLE, cls.HEALTHCHECK_FAILING]
+
+
+def get_replica_state(pod: KubernetesPodV2) -> ReplicaState:
+    phase = pod.phase
+    state = ReplicaState.UNKNOWN
+    if phase is None or not pod.scheduled:
+        state = ReplicaState.UNSCHEDULED
+    elif pod.delete_timestamp:
+        state = ReplicaState.TERMINATING
+    elif phase == "Pending":
+        if not pod.containers or all([c.state == "waiting" for c in pod.containers]):
+            state = ReplicaState.STARTING
+        else:
+            state = ReplicaState.UNHEALTHY
```

Comment:
I _think_ `Pending` means unscheduled?

Reply:
Oddly I did hit the 'Starting' state during some tests with paasta-contract-monitor, so I think unscheduled is still separate from 'phase=Pending'.  I think a pod can be scheduled to a node, but containers still haven't spun up yet (e.g. the short time the node is pulling images perhaps)